### PR TITLE
Add lockInfo tests

### PR DESCRIPTION
**Pull request by PatersonProjects**
This PR adds the tests for the lockInfo diagnostic command. Most of the tests are very straightforward, but response validation requires a lock in place to ensure fields can be checked and so is marked as no-parallel.

Ref: Issue #43, #443

### CODE REVIEW

File: documentdb_tests/compatibility/tests/system/diagnostic/commands/lockInfo/test_lockInfo_core_behavior.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+from documentdb_tests.framework.parametrize import pytest_params
+from documentdb_tests.framework.property_checks import (  # IsType used in CORE_BEHAVIOR_TESTS
+    Eq,
+    IsType,
+)
+
+pytestmark = pytest.mark.admin
+
+
+CORE_BEHAVIOR_TESTS: list[DiagnosticTestCase] = [
```
Comment (fanyangv) [inline documentdb_tests/compatibility/tests/system/diagnostic/commands/lockInfo/test_lockInfo_core_behavior.py]: The parametrized cases in test_lockInfo_core_behavior (returns_ok and returns_lockInfo_array) are identical to test_lockInfo_response_structure (ok_field and lockInfo_is_array). Consider removing the duplicates from one file — they belong in response_structure since they validate response shape.

File: documentdb_tests/compatibility/tests/system/diagnostic/commands/lockInfo/test_lockInfo_errors.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    UNAUTHORIZED_ERROR,
+    UNRECOGNIZED_COMMAND_FIELD_ERROR,
+)
+from documentdb_tests.framework.executor import execute_admin_command, execute_command
+from documentdb_tests.framework.parametrize import pytest_params
+
+pytestmark = pytest.mark.admin
+
+
+ERROR_TESTS: list[DiagnosticTestCase] = [
```
Comment (fanyangv) [inline documentdb_tests/compatibility/tests/system/diagnostic/commands/lockInfo/test_lockInfo_errors.py:24]: No test for case-mismatched command name (e.g., {"LockInfo": 1} or {"LOCKINFO": 1}). The hostInfo PR includes such a test. Consider adding one for consistency.

File: documentdb_tests/compatibility/tests/system/diagnostic/commands/lockInfo/test_lockInfo_argument_validation.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+
+@pytest.mark.parametrize("bson_type,sample_value,spec", ACCEPTANCE_TESTS)
+def test_lockInfo_accepts_any_type(collection, bson_type, sample_value, spec):
+    """Verify lockInfo succeeds when the command field value is a given BSON type."""
+    result = execute_admin_command(collection, {"lockInfo": sample_value})
+    assertProperties(result, {"ok": Eq(1.0)}, msg=spec.msg, raw_res=True)
+
+
+INT_EDGE_TESTS: list[DiagnosticTestCase] = [
```
Comment (fanyangv) [inline documentdb_tests/compatibility/tests/system/diagnostic/commands/lockInfo/test_lockInfo_argument_validation.py]: For consistency with the sibling ping PR (#616) which uses NUMERIC_EDGE_CASES covering -1, 0, and FLOAT_INFINITY, consider adding FLOAT_INFINITY here as well. Both commands ignore their value, and infinity is a special double value beyond what the BSON type acceptance test covers (3.14).

**Review by fanyangv — CHANGES_REQUESTED** [review]

File: documentdb_tests/compatibility/tests/system/diagnostic/commands/lockInfo/test_lockInfo_core_behavior.py (hunk 1 above)
Comment (PatersonProjects) [reply documentdb_tests/compatibility/tests/system/diagnostic/commands/lockInfo/test_lockInfo_core_behavior.py]: Removed duplicates, moved final core_behavior test to argument validation and removed file

File: documentdb_tests/compatibility/tests/system/diagnostic/commands/lockInfo/test_lockInfo_argument_validation.py (hunk 3 above)
Comment (PatersonProjects) [reply documentdb_tests/compatibility/tests/system/diagnostic/commands/lockInfo/test_lockInfo_argument_validation.py]: Added infinity edge case

File: documentdb_tests/compatibility/tests/system/diagnostic/commands/lockInfo/test_lockInfo_errors.py (hunk 2 above)
Comment (PatersonProjects) [reply documentdb_tests/compatibility/tests/system/diagnostic/commands/lockInfo/test_lockInfo_errors.py:24]: Added case-mismatched LockInfo error test

**Review by fanyangv — APPROVED** [review]

**Review by eerxuan — APPROVED** [review]
